src/os_posix/os_fs.c

```diff
@@ -420,9 +431,60 @@ __posix_file_read(
               "%s: handle-read: pread: failed to read %" WT_SIZET_FMT " bytes at offset %" PRIuMAX,
               file_handle->name, chunk, (uintmax_t)offset);
     }
+    WT_STAT_CONN_INCRV(session, block_byte_read_syscall, len);
     return (0);
 }
 
+/*
+ * __posix_file_read_mmap --
+ *     Get the buffer from the mapped region.
+ */
+static int
+__posix_file_read_mmap(
+  WT_FILE_HANDLE *file_handle, WT_SESSION *wt_session, wt_off_t offset, size_t len, void *buf)
+{
+    WT_FILE_HANDLE_POSIX *pfh;
+    WT_SESSION_IMPL *session;
+
+    session = (WT_SESSION_IMPL *)wt_session;
+    pfh = (WT_FILE_HANDLE_POSIX *)file_handle;
+
+    __wt_verbose(session, WT_VERB_READ,
+      "read-mmap: %s, fd=%d, offset=%" PRId64
+      ","
+      "len=%" PRIu64 ", mapped buffer: %p, mapped size = %" PRIu64 "\n",
+      file_handle->name, pfh->fd, offset, (uint64_t)len, (void *)pfh->mmap_buf,
+      (uint64_t)pfh->mmap_size);
+
+    WT_ASSERT(session, S2C(session)->mmap);
```

Comment:
That looks wrong -- if we turn mmap off at some point, we don't reset the read function pointer, do we?  Don't we end up here and assert?

Reply:
Yeah, I did not consider the case where we turn off mmap on an open connection. I assumed that since mmap a connection property, it'll be either  on or off as long as the connection is open.

Reply:
I don't feel strongly, but we'd certainly have to disallow reconfiguring the `mmap` options, which might be more effort than making reconfiguration work.

Reply:
The way I have it now is that we check this option when we open the file. So in order for the config change to take effect, the file would have to be closed and reopen. I don't see any other elegant way to reconfigure.